apps/research-service/src/validations/validators/estimated-area.validator.ts

```diff
@@ -0,0 +1,198 @@
+        upperBoundSite: siteData.upperBoundSite,
+        lowerBoundProject: projectData.lowerBoundProject,
+        upperBoundProject: projectData.upperBoundProject
+      }
+    };
+  }
+
+  async validatePolygons(polygonUuids: string[]): Promise<PolygonValidationResult[]> {
+    const results: PolygonValidationResult[] = [];
+
+    for (const polygonUuid of polygonUuids) {
+      try {
+        const result = await this.validatePolygon(polygonUuid);
+        results.push({
+          polygonUuid,
+          valid: result.valid,
+          extraInfo: result.extraInfo
+        });
+      } catch (error) {
+        results.push({
+          polygonUuid,
+          valid: false,
+          extraInfo: { error: error instanceof Error ? error.message : "Unknown error" }
+        });
+      }
+    }
+
+    return results;
+  }
+
+  private async getSitePolygonWithAssociations(polygonUuid: string): Promise<SitePolygon | null> {
+    return await SitePolygon.findOne({
+      where: { polygonUuid, isActive: true },
+      include: [
+        {
+          model: Site,
+          as: "site",
+          attributes: ["hectaresToRestoreGoal", "projectId"]
+        }
+      ]
+    });
+  }
+
+  private async generateAreaDataSite(sitePolygon: SitePolygon): Promise<{
+    valid: boolean;
+    sumAreaSite: number | null;
+    percentageSite: number | null;
+    totalAreaSite: number | null;
+    lowerBoundSite: number | null;
+    upperBoundSite: number | null;
+  }> {
+    const site = sitePolygon.site != null ? sitePolygon.site : await sitePolygon.loadSite();
+    if (site == null || site.hectaresToRestoreGoal == null || site.hectaresToRestoreGoal <= 0) {
+      return {
+        valid: false,
+        sumAreaSite: null,
+        percentageSite: null,
+        totalAreaSite: site != null ? site.hectaresToRestoreGoal : null,
+        lowerBoundSite: null,
+        upperBoundSite: null
+      };
+    }
+
+    const sumAreaSite = await this.calculateSiteAreaSum(site.uuid);
+    const lowerBound = EstimatedAreaValidator.LOWER_BOUND_MULTIPLIER * site.hectaresToRestoreGoal;
+    const upperBound = EstimatedAreaValidator.UPPER_BOUND_MULTIPLIER * site.hectaresToRestoreGoal;
+    const valid = sumAreaSite >= lowerBound && sumAreaSite <= upperBound;
+    const percentage = (sumAreaSite / site.hectaresToRestoreGoal) * 100;
+
+    return {
+      valid,
+      sumAreaSite: Math.round(sumAreaSite),
+      percentageSite: Math.round(percentage),
+      totalAreaSite: site.hectaresToRestoreGoal,
+      lowerBoundSite: lowerBound,
+      upperBoundSite: upperBound
+    };
+  }
+
+  private async generateAreaDataProject(sitePolygon: SitePolygon): Promise<{
+    valid: boolean;
+    sumAreaProject: number | null;
+    percentageProject: number | null;
+    totalAreaProject: number | null;
+    lowerBoundProject: number | null;
+    upperBoundProject: number | null;
+  }> {
+    const site = sitePolygon.site != null ? sitePolygon.site : await sitePolygon.loadSite();
```

Comment:
The `loadSite` method does these checks for you, so this can simply be
```
const site = await sitePolygon.loadSite();
```
If it's already been loaded, it'll return immediately.